src/test/java/com/endava/internship/collections/StudentSetTest.java

```diff
@@ -0,0 +1,217 @@
+import java.util.*;
+import java.time.LocalDate;
+
+public class StudentSetTest {
+    private StudentSet set;
+
+    private final static Student st1 = new Student("John", LocalDate.of(2000, 5, 12), "");
+
+    private final static Student st2 = new Student("David", LocalDate.of(2001, 6, 20), "");
+
+    private final static Student st3 = new Student("Dmitrii", LocalDate.of(1996, 10, 16), "");
+
+    @BeforeEach
+    void setUpStudentSet(){
+        set = new StudentSet();
+    }
+
+    @Test
+    void addElementShouldReturnTrue(){
+        addElementsToSet(set, st1);
+
+        assertAll(
+                () -> assertThat(set).isNotEmpty(),
+                () -> assertThat(set).hasSize(1)
+        );
+    }
+
+    @Test
+    void addDuplicatedNullToSetShouldAllowASingleNullElement(){
+        assertTrue(set.add(null));
+        assertFalse(set.add(null));
+
+        assertAll(
+                () -> assertThat(set).hasSize(1),
+                () -> assertThat(set).containsNull()
+        );
+    }
+
+    @Test
+    void addDublicatedObjectToSetShouldNotBeAdded(){
+        addElementsToSet(set, st1);
+
+        assertAll(
+                () -> assertFalse(set.add(st1)),
+                () -> assertThat(set).containsOnlyOnce(st1)
+        );
+    }
+
+    @Test
+    void addNullToSetShouldBeAdded(){
+        assertTrue(set.add(null));
+    }
+
+    @Test
+    void setShouldBeEmptyIfNoContainsElements(){
+        assertThat(set).isEmpty();
+    }
+
+    @Test
+    void sizeShouldBe0IfNoContainsElements(){
+        assertEquals(0, set.size());
+    }
+
+    @Test
+    void ifElementIsContainedItShouldReturnTrue(){
+        addElementsToSet(set, st1);
+        assertTrue(set.contains(st1));
+    }
+
+    @Test
+    void ifElementIsNotContainedItShouldReturnFalse(){
+        assertFalse(set.contains(st1));
+    }
+
+    @Test
+    void setShouldBeClearedIfContainsElements(){
+        addElementsToSet(set, st1);
+        set.clear();
+
+        assertAll(
+                () -> assertThat(set).isEmpty(),
+                () -> assertThat(set).hasSize(0)
+        );
+    }
+
+    @Test
+    void elementShoulBeRemovedFromSetIfItIsContained(){
+        assertTrue(set.add(st1));
+
+        assertAll(
+                () -> assertTrue(set.remove(st1)),
+                () -> assertTrue(!set.contains(st1))
+        );
+    }
+
+    @Test
+    void elementShouldBeNotRemovedFromSetIfItIsNotContained(){
+        assertFalse(set.remove(st1));
+    }
+
+    @Test
+    void removeNullElementThatIsNotContainedShouldReturnFalse(){
+        assertFalse(set.remove(null));
+    }
+
+    @Test
+    void nullElementShouldBeRemovedIfItIsContained(){
+        assertTrue(set.add(null));
+        assertTrue(set.remove(null));
+        assertFalse(set.contains(null));
+    }
+
+    @Test
+    void methodShouldReturnIteratorThatContainsNotNullElementsFromTheSet(){
+        addElementsToSet(set, st1, st2);
+        Iterator it = set.iterator();
```

Comment:
**final**  Iterator it = set.iterator();

Reply:
Modified